pkg/kn/commands/revision_list_test.go

```diff
@@ -0,0 +1,166 @@
+	action client_testing.Action, output []string, err error) {
+	buf := new(bytes.Buffer)
+	fakeServing := &fake.FakeServingV1alpha1{&client_testing.Fake{}}
+	cmd := NewKnCommand(KnParams{
+		Output:         buf,
+		ServingFactory: func() (serving.ServingV1alpha1Interface, error) { return fakeServing, nil },
+	})
+	fakeServing.AddReactor("*", "*",
+		func(a client_testing.Action) (bool, runtime.Object, error) {
+			action = a
+			if action.Matches("list", "routes") {
+				return true, routes, nil
+			}
+			return true, revisions, nil
+		})
+
+	cmd.SetArgs(args)
+	err = cmd.Execute()
+	if err != nil {
+		return
+	}
+	output = strings.Split(buf.String(), "\n")
+	return
+}
+
+func TestRevisionListEmpty(t *testing.T) {
+	action, output, err := fakeRevisionList(
+		[]string{"revision", "list"},
+		&v1alpha1.RevisionList{},
+		&v1alpha1.RouteList{})
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	expected := []string{"No resources found.", ""}
+	for i, s := range output {
+		if s != expected[i] {
+			t.Errorf("%d Bad output line %v", i, s)
+		}
+	}
+	if action == nil {
+		t.Errorf("No action")
+	} else if !action.Matches("list", "routes") {
+		t.Errorf("Bad action %v", action)
+	}
+}
+
+var revisionType = metav1.TypeMeta{
+	Kind:       "revision",
+	APIVersion: "serving.knative.dev/v1alpha1",
+}
+var routeType = metav1.TypeMeta{
+	Kind:       "route",
+	APIVersion: "serving.knative.dev/v1alpha1",
+}
+
+func TestRevisionListDefaultOutput(t *testing.T) {
+	foo_label := make(map[string]string)
+	bar_label := make(map[string]string)
+	foo_label["serving.knative.dev/service"] = "f1"
+	bar_label["serving.knative.dev/service"] = "b1"
+
+	// sample RevisionList
+	rev_list := &v1alpha1.RevisionList{
+		Items: []v1alpha1.Revision{
+			v1alpha1.Revision{
+				TypeMeta: revisionType,
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "foo",
+					Labels: foo_label,
+				},
+			},
+			v1alpha1.Revision{
+				TypeMeta: revisionType,
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "bar",
+					Labels: bar_label,
+				},
+			},
+		},
+	}
+	// sample RouteList
+	route_list := &v1alpha1.RouteList{
+		Items: []v1alpha1.Route{
+			v1alpha1.Route{
+				TypeMeta: routeType,
+				Status: v1alpha1.RouteStatus{
+					Domain: "foo.default.example.com",
+					Traffic: []v1alpha1.TrafficTarget{
+						v1alpha1.TrafficTarget{
+							RevisionName: "foo",
+							Percent:      100,
+						},
+					},
+				},
+			},
+			v1alpha1.Route{
+				TypeMeta: routeType,
+				Status: v1alpha1.RouteStatus{
+					Domain: "bar.default.example.com",
+					Traffic: []v1alpha1.TrafficTarget{
+						v1alpha1.TrafficTarget{
+							RevisionName: "bar",
+							Percent:      100,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	action, output, err := fakeRevisionList(
+		[]string{"revision", "list"},
+		rev_list,
+		route_list)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// each line's tab/spaces are replaced by comma
+	expected := []string{"NAME,SERVICE,AGE,TRAFFIC",
```

Comment:
is there a test with multiple routes for one revision?

Reply:
@cppforlife: updated tests to reflect same